### PR TITLE
Run compliance scanner tests against inspec master

### PR DESCRIPTION
**Pull request by phiggins**
In order to have better awareness of breaking changes in the integration of automate and inspec, this PR adds a buildkite task to build the latest inspec from master and use that to run the inspec focused tests.

### CODE REVIEW

File: .expeditor/verify.pipeline.yml [hunk 1]
```diff
@@ -587,6 +587,34 @@ steps:
...
+      LANG: en_US.UTF-8
+    timeout_in_minutes: 20
+    expeditor:
+      executor:
+        linux:
+          single-use: true
+          privileged: true
+    plugins:
+      - gopath-checkout#v1.0.1:
+          import: github.com/chef/automate
```
Comment (phiggins) [inline .expeditor/verify.pipeline.yml]: This is a copy of the existing task for the scanner tests that calls a different script to install inspec.

File: components/compliance-service/Makefile [hunk 2]
```diff
@@ -464,7 +464,7 @@ test: generate-ruby-grpc wait-for-compliance-service
 test-prep: $(MARKET_PATH) $(PROFILES_PATH) start-ssh-node download-sample-market-profiles install-inspec start-es-pg
 
 install-inspec:
-	@inspec --version | grep -q $(shell cat ../../INSPEC_VERSION) || gem install inspec-bin --no-document --version $(shell cat ../../INSPEC_VERSION)
+	@inspec --version 2> /dev/null | grep -q $(shell cat ../../INSPEC_VERSION) || gem install inspec-bin --no-document --version $(shell cat ../../INSPEC_VERSION)
```
Comment (phiggins) [inline components/compliance-service/Makefile]: This change prevents the "inspec: command not found" message when you don't already have inspec installed.

**Comment by phiggins** [comment]
Should this be allowed to fail without failing the entire build?

**Comment by vjeffrey** [comment]
> Should this be allowed to fail without failing the entire build?

My opinion: yes!
Reasoning: if the whole build fails based on this failure, the entire Automate team will be blocked from merging until the inspec issue is fixed. While I believe some gates of this kind are super useful/needed, they should not be at the merge level, they should be at the release level.

File: .expeditor/verify.pipeline.yml [hunk 3]
```diff
@@ -587,6 +587,34 @@ steps:
       - gopath-checkout#v1.0.1:
           import: github.com/chef/automate
 
+  - label: "compliance-service scanner with inspec master"
+    command:
```
Comment (vjeffrey) [inline .expeditor/verify.pipeline.yml]: add `soft_fail: true` ?

File: scripts/install_inspec_master.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+install_inspec() {
+  hab pkg install core/gcc --binlink --force
+  hab pkg install core/ruby --binlink --force
+
+  tmp_dir=$(mktemp -d)
+  git clone --depth=1 https://github.com/inspec/inspec "$tmp_dir"
+  cd "$tmp_dir"
+  bundle
+  bundle exec rake install
```
Comment (zenspider) [inline scripts/install_inspec_master.sh]: This looks great. I have some scripts in inspec/inspec's expeditor dir to make it really easy to cache bundle artifacts. We can get that on another iteration. Cuts down ~3 minutes per. Maybe it's also possible to cache the hab pkgs?

File: .expeditor/verify.pipeline.yml (hunk 3 above)
Comment (zenspider) [reply .expeditor/verify.pipeline.yml]: For now, totally... but at some point, once trust is built up and the system is stabilized, I'd like that removed and eyeballs to hit it when it goes red.

**Pull request merged**